From 95f42ede8c6ce3a552c20872d7bb4fab57298629 Mon Sep 17 00:00:00 2001
From: Nicolai Buchwitz <nb@tipi-net.de>
Date: Sat, 25 Jul 2026 23:46:32 +0200
Subject: [PATCH] pinctrl: rp1: Fix node leak and mapping check in probe

Drop the leaked rp1_node reference and test of_iomap() for NULL instead
of IS_ERR(), which never matches its return value.

Fixes: df868dcd8b11 ("pinctrl: Reinstate the downstream pinctrl-rp1 driver")
Signed-off-by: Nicolai Buchwitz <nb@tipi-net.de>
---
 drivers/pinctrl/pinctrl-rp1.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

--- a/drivers/pinctrl/pinctrl-rp1.c
+++ b/drivers/pinctrl/pinctrl-rp1.c
@@ -1627,11 +1627,10 @@ static int rp1_pinctrl_probe(struct plat
 	else if (pace_pin_updates &&
 		 of_device_is_compatible(rp1_node->parent, "brcm,bcm2712-pcie")) {
 		pc->dummy_base = of_iomap(rp1_node->parent, 0);
-		if (IS_ERR(pc->dummy_base)) {
+		if (!pc->dummy_base)
 			dev_warn(&pdev->dev, "could not map bcm2712 root complex registers\n");
-			pc->dummy_base = NULL;
-		}
 	}
+	of_node_put(rp1_node);
 
 	for (i = 0; i < RP1_NUM_BANKS; i++) {
 		const struct rp1_iobank_desc *bank = &rp1_iobanks[i];
